modin/core/execution/dispatching/factories/dispatcher.py

```diff
@@ -155,152 +156,148 @@ def _update_factory(cls, _):
     @classmethod
     @_inherit_docstrings(factories.BaseFactory._from_pandas)
     def from_pandas(cls, df):
-        return cls.__factory._from_pandas(df)
+        return cls.get_factory()._from_pandas(df)
```

Comment:
we have explicit calls to `Engine.subscribe()` in our I/O methods like `modin.pandas.read_csv()`, it feels like we should remove it from there as `.get_factory()` is now initializing the engine for us